### PR TITLE
Show proto fields for protobuf based web actions

### PR DESCRIPTION
**Pull request by benaskins**
When an action's request type or response type is a proto show the field definitions for that proto.

Before:

Misk Web Actions - Google Docs 2021-04-22 11-52-37

After:

Misk Web Actions - Google Docs 2021-04-22 11-53-10

### CODE REVIEW

**Review by r3mariano — APPROVED** [review]

File: misk/web/tabs/web-actions/src/rewrite/WebActionCardBody.tsx [hunk 1]
```diff
@@ -3,8 +3,9 @@ import { Menu, H5, UL } from "@blueprintjs/core"
 import { WebActionMetadata } from "./types"
 import WebActionCollapse from "./WebActionCollapse"
 import WebActionSendRequest from "./WebActionSendRequest"
-import WebActionParameters from "./WebActionParameters";
-import WebActionDescription from "./WebActionDescription";
+import WebActionRequestParameters from "./WebActionRequestParameters"
+import WebActionDescription from "./WebActionDescription"
+import WebActionResponseType from "./WebActionResponseType";
```
Comment (adrw) [inline misk/web/tabs/web-actions/src/rewrite/WebActionCardBody.tsx:8]: nit: rm semicolon

File: misk/src/main/kotlin/misk/web/metadata/webaction/WebActionMetadata.kt [hunk 2]
```diff
@@ -69,7 +72,9 @@ data class WebActionMetadata(
     },
     requestType = requestType.toString(),
     returnType = returnType.toString(),
```
Comment (adrw) [inline misk/src/main/kotlin/misk/web/metadata/webaction/WebActionMetadata.kt:74]: Could be wrong, but is return type == response type? And I think the return type is already included in the `types` map so the `responseTypes` map would be duplicative

**Review by adrw — COMMENTED** [review]
Curious if this can be done with the existing `types` map instead of adding a duplicate `responseTypes`.

File: misk/web/tabs/web-actions/src/rewrite/WebActionCardBody.tsx (hunk 1 above)
Comment (benaskins) [reply misk/web/tabs/web-actions/src/rewrite/WebActionCardBody.tsx:8]: I'm finding that IntelliJ doesn't format the code in the way I'd expect when automatically importing. I probably need to set some options here. Do you have any tips?

File: misk/src/main/kotlin/misk/web/metadata/webaction/WebActionMetadata.kt (hunk 2 above)
Comment (benaskins) [reply misk/src/main/kotlin/misk/web/metadata/webaction/WebActionMetadata.kt:74]: Unfortunately not. `returnType` will be set to the literal Kotlin type return by the function that implements the action. For example `misk.web.Response<com.squareup.exemplar.protos.HelloWebResponse>`. For the response type I'm reflecting on the `KType` to determine the classified class, in this case `com.squareup.exemplar.protos.HelloWebResponse`.

Currently `types` only includes the types for the request parameter annotated with `@RequestBody` - it doesn't include types for the response.

File: misk/web/tabs/web-actions/src/rewrite/WebActionCardBody.tsx (hunk 1 above)
Comment (benaskins) [reply misk/web/tabs/web-actions/src/rewrite/WebActionCardBody.tsx:8]: I think I've got this sorted now. Thanks for the pickup.

**Pull request merged**